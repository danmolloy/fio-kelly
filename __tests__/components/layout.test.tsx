import "@testing-library/jest-dom";
import { act, fireEvent, render, screen } from "@testing-library/react";
import Layout from "../../components/layout";

describe("<Layout />", () => {
  beforeEach(() => {
    render(
      <Layout>
        <div data-testid="mock-children"></div>
      </Layout>
    );
  });
  it("layout-div is in the document", () => {
    const layout = screen.getByTestId("layout-div");
    expect(layout).toBeInTheDocument();
  });
  it("<Header /> is in the document", () => {
    const header = screen.getByTestId("page-header");
    expect(header).toBeInTheDocument();
  });
  it("<Menu /> renders on menuIcon click", () => {
    const menuIcon = screen.getByTestId("menu-icon");
    act(() => {
      fireEvent.click(menuIcon);
    });
    const menuDiv = screen.getByTestId("menu-div");
    expect(menuDiv).toBeInTheDocument();
  });
  it("{children} is in the document", () => {
    const mockChildren = screen.getByTestId("mock-children");
    expect(mockChildren).toBeInTheDocument();
  });
  it("<Footer /> is in the document", () => {
    const footerSection = screen.getByTestId("footer-section");
    expect(footerSection).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const layout = screen.getByTestId("layout-div");
    expect(layout).toMatchSnapshot();
  });
});
