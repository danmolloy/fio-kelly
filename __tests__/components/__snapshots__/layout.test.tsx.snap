// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> matches snapshot 1`] = `
<div
  data-testid="layout-div"
>
  <div
    class="font-display  transition duration-500 ease-out fixed fill-white  z-30 h-16 w-screen top-0 flex flex-row  items-center justify-between   text-zinc-100"
    data-testid="page-header"
  >
    <h1
      class="md:hidden p-4 "
    >
      Fiona Kelly
    </h1>
    <div
      class="hidden md:flex w-screen  flex-row  justify-end pr-4"
    >
      <button
        class="my-6 p-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
      >
        Home
      </button>
      <a
        class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
        data-testid="About"
        href="/"
      >
        <button
          class="p-2"
        >
          About
        </button>
      </a>
      <a
        class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
        data-testid="Listen"
        href="/"
      >
        <button
          class="p-2"
        >
          Listen
        </button>
      </a>
      <a
        class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
        data-testid="Gallery"
        href="/"
      >
        <button
          class="p-2"
        >
          Gallery
        </button>
      </a>
      <a
        class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
        data-testid="Contact"
        href="/"
      >
        <button
          class="p-2"
        >
          Contact
        </button>
      </a>
    </div>
    <button
      class=" m-2 hover:fill-amber-500 active:fill-amber-400  md:hidden"
      data-testid="menu-icon"
    >
      <svg
        class="m-2"
        height="22"
        width="30"
      >
        <rect
          height="2"
          rx="2"
          style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
          width="30"
        />
        <rect
          height="2"
          rx="2"
          style="opacity: 1;"
          width="30"
          y="8"
        />
        <rect
          height="2"
          rx="2"
          style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
          width="30"
          y="17"
        />
      </svg>
      <p
        class="hidden"
      >
        Menu
      </p>
    </button>
  </div>
  <div
    class="fixed bottom-0 left-0 w-screen min-h-screen bg-black z-20 flex flex-col rounded-tl  text-2xl transition-transform duration-500 ease-in-out translate-y-full"
    data-testid="menu-div"
  >
    <div
      class="flex flex-col w-full justify-evenly font-display "
    >
      <div
        class="w-full h-32"
      />
      <a
        class=" flex flex-col items-center justify-center transform transition duration-700 ease-in-out
          translate-y-12 opacity-0
           "
        href="/"
      >
        <button
          class="px-12 w-full py-4 m-4 flex flex-col text-white  items-end justify-center hover:cursor-pointer hover:text-amber-500 active:text-amber-400"
        >
          About
        </button>
        <div
          class="h-0.5 bg-white mt-2 transition-all duration-500 ease-out self-start mx-8 w-0"
          style="transition-delay: 200ms;"
        />
      </a>
      <a
        class=" flex flex-col items-center justify-center transform transition duration-700 ease-in-out
          translate-y-12 opacity-0
           "
        href="/"
      >
        <button
          class="px-12 w-full py-4 m-4 flex flex-col text-white  items-end justify-center hover:cursor-pointer hover:text-amber-500 active:text-amber-400"
        >
          Listen
        </button>
        <div
          class="h-0.5 bg-white mt-2 transition-all duration-500 ease-out self-start mx-8 w-0"
          style="transition-delay: 300ms;"
        />
      </a>
      <a
        class=" flex flex-col items-center justify-center transform transition duration-700 ease-in-out
          translate-y-12 opacity-0
           "
        href="/"
      >
        <button
          class="px-12 w-full py-4 m-4 flex flex-col text-white  items-end justify-center hover:cursor-pointer hover:text-amber-500 active:text-amber-400"
        >
          Gallery
        </button>
        <div
          class="h-0.5 bg-white mt-2 transition-all duration-500 ease-out self-start mx-8 w-0"
          style="transition-delay: 400ms;"
        />
      </a>
      <a
        class=" flex flex-col items-center justify-center transform transition duration-700 ease-in-out
          translate-y-12 opacity-0
           "
        href="/"
      >
        <button
          class="px-12 w-full py-4 m-4 flex flex-col text-white  items-end justify-center hover:cursor-pointer hover:text-amber-500 active:text-amber-400"
        >
          Contact
        </button>
        <div
          class="h-0.5 bg-white mt-2 transition-all duration-500 ease-out self-start mx-8 w-0"
          style="transition-delay: 500ms;"
        />
      </a>
    </div>
  </div>
  <div>
    <div
      data-testid="mock-children"
    />
  </div>
  <section
    class="z-10 h-24 bottom-0 flex flex-row items-center justify-center border-t border-t-zinc-300 bg-zinc-100"
    data-testid="footer-section"
  >
    <div
      class="flex flex-col items-center justify-center"
    >
      <button
        class="hover:underline font-display text-sm font-light hover:cursor-pointer active:text-zinc-400"
      >
        Back to top
      </button>
      <a
        class="text-sm font-light font-display self-end mt-3 text-gray-600 underline hover:text-blue-500"
        href="https://www.purpleraspberry.uk/"
        target="_blank"
      >
        Purple Rasberry Photography
      </a>
    </div>
  </section>
</div>
`;
